backstop_data/backstop_test.js

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const scenarioConfigs = require('./backstop_scenarios');
+const scenarios = [];
+const excludedScenarios = new Set(['@spectrum-css/circleloader', '@spectrum-css/coachmark', 'index', 'getting-started']);
+const themes = new Set().add('light');
+const scales = new Set().add('medium');
+const packageNameSet = new Set();
+const packageDependentMap = new Map(require('./packageDependentMap'));
+
+let isDocker = false;
+let env = 'local';
+let host = 'host.docker.internal';
+let report = 'ci';
+let captureLimit = 1;
+
+// Shared scenario configuration
+const baseScenarioConfig = {
+  referenceUrl: '',
+  readyEvent: '',
+  readySelector: '',
+  delay: 0,
+  hideSelectors: [],
+  removeSelectors: [],
+  hoverSelector: '',
+  clickSelector: '',
+  postInteractionWait: 0,
+  selectors: ['.spectrum-CSSExample-example'],
+  selectorExpansion: true,
+  expect: 0,
+  misMatchThreshold: 0.1,
+  requireSameDimensions: true
+};
+
+const [...rest] = process.argv.slice(3); // Exclude 'node', 'backstop' and backstop sub command like 'test/approve'
+
+// Process the arguments
+rest.forEach(argv => {
+  if (argv === '--moby') {
+    isDocker = true;
+  } else if (argv.startsWith('--env=')) {
+    env = argv.slice('--env='.length);
+  } else if (argv.startsWith('themes=')) {
+    themes.clear();
+    argv.slice('themes='.length).split(',').forEach(t => themes.add(t));
+  } else if (argv.startsWith('scales=')) {
+    scales.clear();
+    argv.slice('scales='.length).split(',').forEach(s => scales.add(s));
+  } else if (!argv.startsWith('--')) {
+    packageNameSet.add(`@spectrum-css/${argv}`);
+    packageDependentMap.get(`@spectrum-css/${argv}`).forEach(i => packageNameSet.add(i));
+  }
+});
+
+// Identify the runtime environment and setup hostname, etc
+if (isDocker === true) {
+  host = (env === 'local' ? 'host.docker.internal' : '127.0.0.1');
```

Comment:
not sure if those ever change. maybe moving them into a global variable and exposing it in the beginning of the script might be better than hard-coding it.